metadata-ingestion/scripts/avro_codegen.py

```diff
@@ -66,29 +65,77 @@ def load_schemas(schemas_path: str) -> Dict[str, dict]:
     return schemas
 
 
-def merge_schemas(schemas_obj: List[Any]) -> str:
-    # Combine schemas.
-    merged = ["null"] + schemas_obj
+def patch_schemas(schemas: Dict[str, dict], pdl_path: Path) -> Dict[str, dict]:
+    urn_arrays: Dict[
+        str, List[Tuple[str, str]]
+    ] = {}  # schema name -> list of (field name, type)
+
+    # First, we need to load the PDL files and find all urn arrays.
+    for pdl_file in Path(pdl_path).glob("**/*.pdl"):
+        pdl_text = pdl_file.read_text()
+        arrays = re.findall(
+            r"^\s*(\w+)\s*:\s*(?:optional\s+)?array\[(\w*Urn)\]",
+            pdl_text,
+            re.MULTILINE,
+        )
```

Comment:
Feel like I'm missing context on this file and when it's used, but this looks like it relies on all our `Urn` schemas (? what's the term here?) having the `Urn` suffix. I guess that's pretty logical, but have we documented this anywhere? At the least, can you add a comment onto the relevant function, cause regexes are hard to parse

Reply:
added a comment about the assumption